src/message-handler/src/NetworkAPIHandler.cpp

```diff
@@ -1,13 +1,17 @@
 #include "NetworkAPIHandler.h"
 
-NetworkAPIHandler::NetworkAPIHandler(IBSP& bsp, ILogger& logger) : m_bsp(bsp), m_logger(logger) {}
+NetworkAPIHandler::NetworkAPIHandler(IBSP& bsp, ILogger& logger, INetworkManager& networkManager) :
+    m_bsp(bsp), m_logger(logger), m_networkManager(networkManager) {}
 
 std::variant<ErrorNum, std::optional<NetworkApiDTO>> NetworkAPIHandler::handleApiCall(
-    const NetworkApiDTO& apiCall) {
+    const MessageDTO& message, const NetworkApiDTO& apiCall) {
     const NetworkApiDTOType& call = apiCall.getApiCall();
-    if (const auto* ipRequest = std::get_if<IPDiscoveryDTO>(&call)) {
-        // Todo: Properly handle call. Return an empty DTO for now.
-        return std::optional<NetworkApiDTO>({});
+    if (const auto* ipDiscovery = std::get_if<IPDiscoveryDTO>(&call)) {
+        if (m_networkManager.registerAgent(message.getSourceId(), ipDiscovery->getIP())) {
```

Comment:
Maybe just pass the sourceId, unless you eventually need to send the whole message eventually or something simillar. Thing is, someone could pass a MessageDTO that isn't the one that contains the NetworkApiDTO

Reply:
Changed